### ISSUE TITLE
Exception Metrics cause IllegalArgumentException when using custom ErrorDecoders

### ISSUE THREAD

**Issue by DarkAtra** [OpenFeign/feign#1466]
Seems like the `MeteredInvocationHandleFactory` in `feign-micrometer` 11.5 tries to create both a timer and counter with the same name when using a custom error decoder that does not return a subclass of `FeignException`. This results in the following exception:

```
Caused by: java.lang.IllegalArgumentException: There is already a registered meter of a different type with the same name
	at io.micrometer.core.instrument.MeterRegistry.registerMeterIfNecessary(MeterRegistry.java:571)
	at io.micrometer.core.instrument.MeterRegistry.registerMeterIfNecessary(MeterRegistry.java:561)
	at io.micrometer.core.instrument.MeterRegistry.counter(MeterRegistry.java:284)
	at io.micrometer.core.instrument.Counter$Builder.register(Counter.java:128)
	at io.micrometer.core.instrument.MeterRegistry.counter(MeterRegistry.java:391)
	at feign.micrometer.MeteredInvocationHandleFactory.createExceptionCounter(MeteredInvocationHandleFactory.java:108)
	at feign.micrometer.MeteredInvocationHandleFactory.lambda$create$0(MeteredInvocationHandleFactory.java:85)
	at com.sun.proxy.$Proxy128.getById(Unknown Source)
```

I have created [this repo](https://github.com/DarkAtra/openfeign-metrics-issue) to demonstrate the issue (refer to the tests).

**Comment by velo** [comment]
Can you make that repo into a PR with a test? Thanks

**Comment by DarkAtra** [comment]
do you have any preferences for the metric name? right now both are named `feign.exception` if i recall correctly

**Comment by velo** [comment]
> do you have any preferences for the metric name? right now both are named `feign.exception` if i recall correctly

If both express the same information, probably is safe to just eliminate one

**Comment by OlgaMaciaszek** [comment]
We've come across this issue while trying to upgrade Spring Cloud OpenFeign to Feign `11.5` so we're waiting for it to get solved for the upgrade.

**Comment by velo** [comment]
Fixed on 11.6
